### DISCUSSION TITLE
Alarm server - automated command execution use cases

### DISCUSSION THREAD

**Discussion by UrbanBobek** [ControlSystemStudio/phoebus#2692] [Q&A] [answered]
Hi all,

I have been playing around with the automated actions for the Alarm server and was wondering what are some of the use cases where the automated commands are used? In our case we are running the Alarm server service on a separate machine inside a docker container, so I am having a hard time seeing where we could use this functionality.

Best regards, 
Urban

**Comment by kasemir** [comment, marked as answer]
You can safely ignore them.
The alarm system should not "do" anything. Its purpose is to help knowledgeable operators do their job, see "Alarm Guidelines" on https://controlssoftware.sns.ornl.gov/training/2022_USPAS/. The "cmd:.." type actions which execute something have no use in typical alarm system operation. If you know exactly what command to execute, for example "Open relieve valve when pressure is too high", you automate that by implementing it via calcout or sequence logic on the IOC, you don't use "cmd:..." actions on the alarm server for that.
The "mailto:" actions can be useful to send emails (or text messages) after some delay like 15 minutes in case nobody is in the control room (but then again who would do anything about those messages?).
See also online help, https://github.com/ControlSystemStudio/phoebus/blob/master/app/alarm/ui/doc/index.rst#automated-actions

**Reply by UrbanBobek** [reply]
All clear!
Thank you for the fast response and for the links.

**Reply by shroffk** [reply]
+1 on **not** doing controls operations using cmd:

I have however found `cmd:` very useful when integrating different notification systems using their command line tools or REST wrappers... prometheus, twitter, slack, etc. Instead on having to add a dedicated action of each of them we can use their respective command line tools installed on the server which can then be invoked by the alarm server using cmd.

**Reply by kasemir** [reply]
Yes, it's for integrating other notification systems, like triggering a script that activates a phone dialer. But that requires a full understanding of that other system and this alarm system, so if you're just getting started, ignore.
